File: hw/vizcaino/u3/hw17/hw17AllConceptsProgramTypeScript/model/StreamingPlatform.ts
import { User } from "./User";
import { PlayableContent } from "./PlayableContent";

export class StreamingPlatform {
    constructor(private name: string) {}

    play(user: User, content: PlayableContent) {
        console.log(
            `${user.getName()} is using ${this.name} (${user.getSubscription().getPlan()} plan)`
        );
        content.play();
    }
}

File: hw/vizcaino/u3/hw17OOPConcepts/hw17AllConceptsProgramPython/model/streaming_platform.py
class StreamingPlatform:

    def __init__(self, name):
        self._name = name

    def play_content(self, user, content):
        print(f"{user.get_name()} is using {self._name} ({user.get_subscription().get_plan_name()} plan)")
        content.play()